### PR TITLE
Remove 'iotjs_handler_t' from the binding layer.

### PR DESCRIPTION
**Pull request by LaszloLango**
Binary size reduced from **309 280** bytes to **305 184** bytes.

### CODE REVIEW

File: src/iotjs_binding.h [hunk 1]
```diff
@@ -251,36 +201,41 @@ iotjs_jval_t iotjs_jval_create_function_with_dispatch(
...
+#define DJHANDLER_GET_REQUIRED_CONF_VALUE(src, target, property, type)       \
+  do {                                                                       \
+    iotjs_jval_t jtmp = iotjs_jval_get_property(src, property);              \
+    if (iotjs_jval_is_undefined(jtmp)) {                                     \
+      return jerry_create_error_sz(TYPE,                                     \
+                                   "Missing argument, required " property);  \
+    } else if (iotjs_jval_is_##type(jtmp))                                   \
+      target = iotjs_jval_as_##type(jtmp);                                   \
+    else {                                                                   \
+      return jerry_create_error_sz(TYPE, "Bad arguments, required " property \
```
Comment (galpeter) [inline src/iotjs_binding.h]: I'm not sure, but shouldn't the `jtmp` be released before returning?

Comment (LaszloLango) [reply src/iotjs_binding.h]: Right, I'll fix it.

**Comment by LaszloLango** [comment]
@hs0225 could you test it on artik053? I cannot run the adc, gpio, i2c, spi and uart tests.

**Comment by glistening** [comment]
@LaszloLango 
@hs0225 had day off this Thurs and Friday. He will be back next Monday.
This PR seems fixing some bug since you added `bug` label. Could you explain what bug was there?

**Comment by LaszloLango** [comment]
@glistening my bad. I wanted to use the `binding` label. :) It is not a bugfix. This PR is related to #1201.

File: src/modules/iotjs_module_udp.c [hunk 2]
```diff
@@ -409,49 +417,61 @@ void SetMembership(iotjs_jhandler_t* jhandler, uv_membership membership) {
...
 
-  iotjs_jhandler_return_null(jhandler);
+  return jerry_create_null();
 #endif
 }
 
 
 JHANDLER_FUNCTION(AddMembership) {
-  SetMembership(jhandler, UV_JOIN_GROUP);
+  JHANDLER_DECLARE_THIS_PTR(udpwrap, udp_wrap);
```
Comment (galpeter) [inline src/modules/iotjs_module_udp.c]: The `AddMemebership` and `DropMembership` methods are almost the same, can't we just pass all jerry arguments to the SetMembership function or create another one which merges the two method's common parts?

**Comment by LaszloLango** [comment]
@galpeter, @glistening, @hs0225 I've updated the PR.

**Note**: After the JerryScript update the binary size was increased and it effects the gain of this patch. Here are the current results:

| branch | binary   | .text | .data | .bss |
|---------|---------|-------|------|------|
| master | 309280 | 305850 | 872 | 265568 |
| remove-iotjs_jhandler_t | 309280 | 303345 | 872 | 265568 |

**Comment by glistening** [comment]
@LaszloLango LGTM (except for adc). @hs0225 Could you review the changed code about `adc`?

**Comment by LaszloLango** [comment]
@hs0225 great! Thanks for the validation.

**Pull request merged**